utils/types.d.ts

```diff
@@ -0,0 +1,23 @@
+export interface IPost {
```

Comment:
are you sure every post have this format only? As per my knowledge, we have `children` prop as well, which is optional. Moreover looks like you have created this type for blog.